spatialmath/sphere.go

```diff
@@ -200,3 +202,38 @@ func sphereInSphere(inner, outer *sphere) bool {
 func sphereInBox(s *sphere, b *box) bool {
 	return -pointVsBoxDistance(b, s.pose.Point()) >= s.radius
 }
+
+// TODO: add function description
+// TODO: double check this still works as expected
+// TODO: add s/o link for where I found this code.
+func (s *sphere) ToPointCloud(options map[string]interface{}) ([]r3.Vector, error) {
+	moveTo := s.pose.Point()
+	phi := math.Pi * (3.0 - math.Sqrt(5.0))
+	nums := 500 * s.radius
+	myList := make([]r3.Vector, int32(nums))
+	for i := 0.; i < nums; i++ {
+		y := 1 - (i/(nums-1))*2
+		radius := math.Sqrt(1 - y*y)
+		theta := phi * i
+		x := (math.Cos(theta)*radius)*s.radius + moveTo.X
+		z := (math.Sin(theta)*radius)*s.radius + moveTo.Z
+		y = y*s.radius + moveTo.Y
+		myVec := r3.Vector{x, y, z}
+		myList = append(myList, myVec)
+	}
+
+	// This is here so I can export myList as a python list
+	// will be deleted
+	lastList := golist.New()
+	for i := 0; i < len(myList); i++ {
+		pointsList := golist.New()
+		pointsList.Append(myList[i].X)
+		pointsList.Append(myList[i].Y)
+		pointsList.Append(myList[i].Z)
+		lastList.Append(pointsList)
+	}
+	f, _ := os.Create("/Users/nick/Desktop/play/last.txt")
```

Comment:
Recommend not ever putting this sort of stuff in commits

Reply:
removed. won't happen again